File: Python/0343.py
class Solution:
    def integerBreak(self, n: int) -> int:
        if n < 4:
            return n-1
        exp = n//3
        match n%3:
            case 0:
                return 3 ** exp
            case 1:
                return 3 ** (exp-1) * 4
            case 2:
                return 3 ** exp * 2

# Time Complexity: O(log(n))
# Space Complexity: O(1)

File: C++/0343.cpp
#include <cmath>

class Solution {
public:
    int integerBreak(int n) {
        if(n < 4) return n-1;
        int exp = n / 3;
        switch(n%3) {
            case 0: return pow(3, exp);
            case 1: return pow(3, (exp-1)) * 4;
            case 2: return pow(3, exp) * 2;
        }
        return 1;
    }
};

int main(){
    Solution sol;
    int n = 10;
    int ans = sol.integerBreak(n);
    return 0;
}

// Time Complexity: O(log(n))
// Space Complexity: O(1)